### PR TITLE
feat(virtctl,create): Add support for sysprep volumes

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

This adds support for specifying a sysprep volume to virtctl create vm by
adding the --sysprep flag. A test case for specifying a sysprep volume is
added to the virtctl create vm functests.

Before this PR:

Users cannot specify a sysprep volume in VMs created with virtctl create vm.

After this PR:

Users can now specify a sysprep volume in VMs created with virtctl create vm.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
virtctl: Users can specify a sysprep volume in VMs created with virtctl create vm
```

### CODE REVIEW

**Review by fossedihelm — COMMENTED** [review]
Looks good overall.
I'm wondering what's the added value of the e2e test actually

File: pkg/virtctl/create/vm/vm.go [hunk 1]
```diff
@@ -63,6 +63,11 @@ const (
 	InferPreferenceFlag        = "infer-preference"
 	InferPreferenceFromFlag    = "infer-preference-from"
 	VolumeImportFlag           = "volume-import"
+	SysprepVolumeFlag          = "volume-sysprep"
+
+	SysprepDisk      = "sysprepdisk"
+	SysprepConfigMap = "configMap"
+	SysprepSecret    = "secret"
```
[the remark this answers is not in the document]
Comment (0xFelix) [reply pkg/virtctl/create/vm/vm.go:70]: To make them reusable in `vm_test.go`.

**Comment by 0xFelix** [comment]
> I'm wondering what's the added value of the e2e test actually

To verify that the VM admitter is happy with the generated manifest. It uses DryRun so it does not create an actual VM.

File: pkg/virtctl/create/vm/vm.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virtctl/create/vm/vm.go:70]: e2e test should not use the productions const.
If something changes in the production code, we have to manually adjust the e2e test as well, without automatically inheriting them.
We should start decoupling them.

[the remark this answers is not in the document]
Comment (0xFelix) [reply pkg/virtctl/create/vm/vm.go:70]: `vm_test.go` is the unit test file of `vm.go`.

[the remark this answers is not in the document]
Comment (enp0s3) [reply pkg/virtctl/create/vm/vm.go:70]: @0xFelix @fossedihelm FYI https://github.com/kubevirt/kubevirt/pull/13058

**Comment by alicefr** [comment]
Looks good
/approve

**Comment by fossedihelm** [comment]
/lgtm
Thanks

**Pull request merged**